File: source/test/java/com/xeomar/annex/LaunchTaskTest.java
package com.xeomar.annex;

import org.junit.Test;

import static org.hamcrest.Matchers.*;
import static org.junit.Assert.assertThat;

public class LaunchTaskTest {

	@Test
	public void testExecute() throws Exception {
		TaskResult result = TaskResult.parse( new Program().runTasksFromString( UpdateTask.LAUNCH + " java" ) );
		assertThat( result.getStatus(), is( TaskStatus.SUCCESS ) );
		assertThat( result.getMessage(), is( "java" ) );
	}

	@Test
	public void testExecuteFailure() throws Exception {
		TaskResult result = TaskResult.parse( new Program().runTasksFromString( UpdateTask.LAUNCH + " invalid" ) );
		assertThat( result.getStatus(), is( TaskStatus.FAILURE ) );
		assertThat( result.getMessage(), startsWith( "IOException: Cannot run program \"invalid\"" ) );
	}

}

File: source/test/java/com/xeomar/annex/LaunchTaskTest.kt
package com.xeomar.annex

import org.hamcrest.Matchers
import org.hamcrest.Matchers.`is`
import org.junit.Assert.assertThat
import org.junit.Test

class LaunchTaskTest {

	@Test
	fun testExecute() {
		val result = TaskResult.parse(Program().runTasksFromString("launch java"))
		assertThat(result.code, `is`(0))
		assertThat(result.message, Matchers.`is`("java"))
	}

	@Test
	fun testExecuteFailure() {
		val result = TaskResult.parse(Program().runTasksFromString("launch invalid"))
		assertThat(result.code, `is`(1))
		assertThat(result.message, Matchers.startsWith("IOException: Cannot run program \"invalid\""))
	}

}